src/native/http2_stream_manager.c

```diff
@@ -0,0 +1,403 @@
+
+        AWS_FATAL_ASSERT(!aws_jni_check_and_clear_exception(env));
+        (*env)->DeleteLocalRef(env, java_http2_stream_manager);
+    }
+
+    // We're done with this wrapper, free it.
+    s_destroy_manager_binding(binding);
+}
+
+JNIEXPORT jlong JNICALL Java_software_amazon_awssdk_crt_http_Http2StreamManager_http2StreamManagerNew(
+    JNIEnv *env,
+    jclass jni_class,
+    jobject stream_manager_jobject,
+    jlong jni_client_bootstrap,
+    jlong jni_socket_options,
+    jlong jni_tls_ctx,
+    jlongArray java_marshalled_settings,
+    jbyteArray jni_endpoint,
+    jint jni_port,
+    jint jni_proxy_connection_type,
+    jbyteArray jni_proxy_host,
+    jint jni_proxy_port,
+    jlong jni_proxy_tls_context,
+    jint jni_proxy_authorization_type,
+    jbyteArray jni_proxy_authorization_username,
+    jbyteArray jni_proxy_authorization_password,
+    jboolean jni_manual_window_management,
+    jlong jni_monitoring_throughput_threshold_in_bytes_per_second,
+    jint jni_monitoring_failure_interval_in_seconds,
+    jint jni_max_conns,
+    jint jni_ideal_concurrent_streams_per_connection,
+    jint jni_max_concurrent_streams_per_connection) {
+
+    (void)jni_class;
+
+    struct aws_client_bootstrap *client_bootstrap = (struct aws_client_bootstrap *)jni_client_bootstrap;
+    struct aws_socket_options *socket_options = (struct aws_socket_options *)jni_socket_options;
+    struct aws_tls_ctx *tls_ctx = (struct aws_tls_ctx *)jni_tls_ctx;
+    struct aws_http2_stream_manager_binding *binding = NULL;
+    struct aws_allocator *allocator = aws_jni_get_allocator();
+
+    if (!client_bootstrap) {
+        aws_jni_throw_runtime_exception(env, "ClientBootstrap can't be null");
+        return (jlong)NULL;
+    }
+
+    if (!socket_options) {
+        aws_jni_throw_runtime_exception(env, "SocketOptions can't be null");
+        return (jlong)NULL;
+    }
+
+    const size_t marshalled_len = (*env)->GetArrayLength(env, java_marshalled_settings);
+    AWS_ASSERT(marshalled_len % 2 == 0);
+
+    size_t num_initial_settings = marshalled_len / 2;
+    struct aws_http2_setting *initial_settings =
+        num_initial_settings ? aws_mem_calloc(allocator, num_initial_settings, sizeof(struct aws_http2_setting)) : NULL;
+
+    jlong *marshalled_settings = (*env)->GetLongArrayElements(env, java_marshalled_settings, NULL);
+    for (size_t i = 0; i < num_initial_settings; i++) {
+        jlong id = marshalled_settings[i * 2];
+        initial_settings[i].id = id;
+        jlong value = marshalled_settings[i * 2 + 1];
+        initial_settings[i].value = (uint32_t)value;
+    }
+
+    struct aws_byte_cursor endpoint = aws_jni_byte_cursor_from_jbyteArray_acquire(env, jni_endpoint);
+
+    if (jni_port <= 0 || 65535 < jni_port) {
+        aws_jni_throw_runtime_exception(env, "Port must be between 1 and 65535");
+        goto cleanup;
+    }
+
+    if (jni_max_conns <= 0) {
+        aws_jni_throw_runtime_exception(env, "Max Connections must be > 0");
```

Comment:
aws_jni_throw_illegal_argument_exception() probably better

or we could validate a lot of this stuff in some common package-private java function before passing it into C